Onboarding feedback

William-436 opened the pull request:
Removed underlines in "Seller_Profile_FeatureStepDefinitions" step definitions file per Onboarding Task 2 feedback.
Made miscellaneous changes prior to receiving feedback.

William-436 committed:
Miscellaneous changes. Do not remember as not in habit of committing anything yet.

William-436 committed:
Included unused, as of now, data sheets

William-436 committed:
Removed underscores from feature file's filename

William-436 committed:
Added constructor to create new page objects

William-436 committed:
Moved initialization of driver to CommonDriver

William-436 committed:
Removed driver from parameter when calling a method

William-436 committed:
Moved login setup and other login components to CommonDriver

William-436 committed:
Implemented POM

amanbansalnz commented on MVPSt_2023_Jan/MVPSt_2023_Jan/Pages/HomePage.cs:
```diff
@@ -1,177 +1,276 @@
             // create object for new SelectElement class
-            IWebElement yearofgradDropDownElement = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[4]/div/div[2]/div/div/div[2]/div[3]/select"));
-            SelectElement selectayearofgrad = new SelectElement(yearofgradDropDownElement);
+            //IWebElement yearofgradDropDownElement = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[4]/div/div[2]/div/div/div[2]/div[3]/select"));
```
remember to remove comments :)

William-436 replied:
I removed many, many comments from all files but probably overlooked a few

amanbansalnz commented on MVPSt_2023_Jan/MVPSt_2023_Jan/Pages/TYSPortalPage.cs:
```diff
@@ -1,45 +1,80 @@
             // find Email address textbox and enter valid Email address
-            IWebElement emailTextbox = driver.FindElement(By.Name("email"));
+            //IWebElement emailTextbox = driver.FindElement(By.Name("email"));
             emailTextbox.SendKeys("wbugan@aol.com");
```
you could pass this through the feature but ill leave that to you to decide

William-436 replied:
I prefer to leave the code as it is in order to keep my logic and structure consistent with my understanding of the concepts.

amanbansalnz commented on MVPSt_2023_Jan/MVPSt_2023_Jan/Utilities/CommonDriver.cs:
```diff
@@ -1,34 +1,50 @@
 
+        // Nunit Hooks
         //[SetUp]
         //[OneTimeSetUp]
```
uncomment the hooks https://codebun.com/how-to-implement-hooks-in-specflow-c/ have look at that link for hooks, once you uncomment this you don't need to call its in your stepDef. See comment on stepDef class

William-436 replied:
Added Specflow [BeforeScenario] to CommonDriver to perform SetupDriver commands

amanbansalnz commented on MVPSt_2023_Jan/MVPSt_2023_Jan/StepDefinitions/SellerProfileFeatureStepDefinitions.cs:
```diff
@@ -1,42 +1,47 @@
-            driver = new ChromeDriver();
-
-            tysportalPageObj.SigninActions(driver);
+            SetupDriver();
```
you don't need to do this as the Hook will execute this for you!

William-436 replied:
Removed call to SetupDriver in Given logic in step definition file

William-436 committed:
Removed many commented-out lines.
Added Specflow [BeforeScenario] hook to CommonDriver to perform SetupDriver.
Removed call to SetupDriver in Given logic in step definition file.
